cli/src/commands/describe.rs

```diff
@@ -67,35 +77,96 @@ pub(crate) fn cmd_describe(
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer).unwrap();
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        let template =
-            description_template_for_describe(ui, command.settings(), &workspace_command, &commit)?;
-        edit_description(workspace_command.repo(), &template, command.settings())?
+        None
     };
-    if description == *commit.description() && !args.reset_author {
-        writeln!(ui.status(), "Nothing changed.")?;
+
+    // TODO: Allow editing description of multiple commits in a single editor.
+    let commit_descriptions: IndexMap<_, _> = commits
+        .iter()
+        .rev()
+        .map(
+            |commit| -> Result<Option<(&CommitId, String)>, CommandError> {
+                let original_description = commit.description();
+                let new_description = if args.no_edit {
+                    commit.description().to_owned()
+                } else if let Some(shared_description) = &shared_description {
+                    shared_description.to_owned()
+                } else {
+                    let template = description_template_for_describe(
+                        ui,
+                        command.settings(),
+                        &workspace_command,
+                        commit,
+                    )?;
+                    edit_description(workspace_command.repo(), &template, command.settings())?
+                };
+                if new_description == *original_description && !args.reset_author {
+                    Ok(None)
+                } else {
+                    Ok(Some((commit.id(), new_description.to_owned())))
+                }
+            },
+        )
+        .filter_map_ok(|x| x)
+        .try_collect()?;
+
+    let mut tx = workspace_command.start_transaction();
+    let tx_description = if commits.len() == 1 {
+        format!("describe commit {}", commits[0].id().hex())
     } else {
-        let mut tx = workspace_command.start_transaction();
-        let mut commit_builder = tx
-            .mut_repo()
-            .rewrite_commit(command.settings(), &commit)
-            .set_description(description);
-        if args.reset_author {
-            let new_author = commit_builder.committer().clone();
-            commit_builder = commit_builder.set_author(new_author);
-        }
-        commit_builder.write()?;
-        tx.finish(ui, format!("describe commit {}", commit.id().hex()))?;
+        format!(
+            "describe commit {} and {} more",
+            commits[0].id().hex(),
+            commits.len() - 1
+        )
+    };
+
+    let mut num_described = 0;
+    let mut num_rebased = 0;
+    tx.mut_repo().transform_descendants(
```

Comment:
can't we simply let `tx.finish()` do rebase? (I assume the rebaser can handle multiple rewrites. Ignore my comment if it can't.)

Reply:
It can handle multiple rewrites, but when editing a commit and its descendant(s), I think this would create an additional entry in the obslog for the descendants. I think we can easily switch to rewriting individual commits one at a time and letting the auto-rebaser handle it if that's preferred.